.github/actions/pylint/index.js

```diff
@@ -0,0 +1,120 @@
+const core = require('@actions/core');
+const exec = require('@actions/exec');
+
+/**
+ * Execute pylint and report issues, if any found.
+ */
+async function run() {
+    let pylintOutput = "";  // Pylint process output.
+    const pylintDisable = core.getInput('pylint-disable') || "";
+    const pylintRCFile = core.getInput('pylint-rcfile') || "";
+    const pylintPaths = core.getInput('pylint-paths');
+
+    try {
+        // Run pylint
+        await exec.exec("pylint", [
+            `--disable=${pylintDisable}`,
+            `--rcfile=${pylintRCFile}`,
+            `--output-format=json`,
+            ...pylintPaths.split(" ")
+        ], {
+            listeners: {
+                stdout: (data) => pylintOutput += data.toString()
+            }
+        });
+    } catch (pylintError) {
+        try {
+            reportResults(JSON.parse(pylintOutput));
+        } catch (reportingError) {
+            // The exception thrown by `exec` doesn't include the exit
+            // code as a field (we'd have to parse the error message).
+            // So report the pylint process error here as well, in case
+            // that was the root cause for `reportResults` failing.
+            console.error(pylintError);
+            console.error(reportingError);
+        }
+    }
+}
+
+/**
+ * Create GitHub annotations and set the action to failed.
+ *
+ * Report a summary followed by per-line annotations.
+ *
+ * @param pylintMessages List of Pylint messages decoded from JSON.
+ */
+function reportResults(pylintMessages) {
+    const pylintErrors = pylintMessages.filter(message => message.type === 'error');
+    const pylintWarnings = pylintMessages.filter(message => message.type === 'warning');
+    const pylintConvention = pylintMessages.filter(message => message.type === 'convention');
+    const pylintRefactor = pylintMessages.filter(message => message.type === 'refactor');
+    const pylintInfo = pylintMessages.filter(message => message.type === 'info');
+
+    const prAnnotationMsg = `\n${pylintErrors.length} error\n\n` +
+        `${pylintWarnings.length} warning\n\n` +
+        `${pylintConvention.length} convention\n\n` +
+        `${pylintRefactor.length} refactor\n\n` +
+        `${pylintInfo.length} info`;
+
+    const prAnnotationProps = {
+        title: "Linting summary"
+    };
+
+    core.setFailed("Pylint linter issues detected");
+
+    if (pylintErrors.length) {
+        core.error(prAnnotationMsg, prAnnotationProps);
+    } else {
+        core.warning(prAnnotationMsg, prAnnotationProps);
+    }
```

Comment:
I considered that, the problem is you don't get to customize the title. But yeah I agree it would be nice to free up an annotation, so will roll them together.

Reply:
Updated as part of e3af06d whilst addressing https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/105#discussion_r758654438